Praveen's WGLC comments (editorial) (#3928)
quicwg/base-drafts: draft-ietf-quic-recovery.md

({{?RFC5681}}, {{?RFC6675}}).  In order to remain similar to TCP,
implementations SHOULD NOT use a packet threshold less than 3; see {{?RFC5681}}.

Some networks may exhibit higher degrees of packet reordering, causing a sender
to detect spurious losses. Additionally, packet reordering could be more common
with QUIC than TCP, because network elements that could observe and fix the

> order of reordered TCP packets cannot do it with QUIC. Algorithms that increase

Reviewer: ```suggestion
order of reordered TCP packets cannot do that for QUIC. Algorithms that increase
```

Author: Missed this. Fixed this in https://github.com/quicwg/base-drafts/commit/8abb9fa2354ae341637ce43479a707844aa4a7cf

persistent congestion is equal to: (1 * kPersistentCongestionThreshold) = 3.
Because the threshold was reached and because none of the packets between the
oldest and the newest packets are acknowledged, the network is considered to
have experienced persistent congestion. Note that this example shows the
occurrence of PTOs, but that is not required for persistent congestion to be

> established.

Reviewer: Won't this conflict with the other change?

Author: Yup, I'll prefer that text when merging.

Packets that carry only ACK frames can be heuristically identified by observing
packet size.  Acknowledgement patterns may expose information about link

> characteristics or application behavior.  To reduce leaked information,
> endpoints can use PADDING frames, at a potential cost to performance, or bundle
> acknowledgments with other frames.

Reviewer 2: ```suggestion
characteristics or application behavior.  To reduce leaked information,
endpoints can bundle acknowledgments with other frames, or add
PADDING frames, at a potential cost to performance.
```
